### PR TITLE
SCT-1251 Add configuration class

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: src/jgikbase/idmapping/config.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    :ivar mongo_host: the host of the MongoDB instance, including the port.
+    :ivar mongo_db: the MongoDB database to use for the ID mapping service.
+    :ivar mongo_user: the username to use with MongoDB, if any.
+    :ivar mongo_pwd: the password to use with MongoDB, if any.
+    :ivar ignore_ip_headers: True if the X-Real-IP and X-Forwarded-For headers should be ignored.
+    """
+
+    ENV_VAR_IDMAPPING = 'ID_MAPPING_CONFIG'
+    """
```
Comment (jayrbolton) [inline src/jgikbase/idmapping/config.py]: Just out of curiosity, do docstrings places after these class vars show up in __doc__, or are they used by sphinx?

Comment (MrCreosote) [reply src/jgikbase/idmapping/config.py]: They show up in sphinx; I've never used `doc` so I can't answer that one.

File: src/jgikbase/idmapping/config.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        """
+        if not cfgfile:
+            cfgfile = self._get_cfg_from_env()
+        cfg = self._get_cfg(cfgfile)
+        self.ignore_ip_headers = self._TRUE == cfg.get(self.KEY_IGNORE_IP_HEADERS)
+        self.mongo_host = self._get_string(self.KEY_MONGO_HOST, cfg)
+        self.mongo_db = self._get_string(self.KEY_MONGO_DB, cfg)
+        self.mongo_user = self._get_string(self.KEY_MONGO_USER, cfg, False)
+        mongo_pwd = self._get_string(self.KEY_MONGO_PWD, cfg, False)
+        if bool(self.mongo_user) ^ bool(mongo_pwd):  # xor
```
Comment (jayrbolton) [inline src/jgikbase/idmapping/config.py]: If the server only actually supports mongo, and both of these are blank, will an error be raised down the line on server bootup?

**Comment by MrCreosote** [comment]
I'll use real files when I do the integration tests.

**Pull request merged**